apps/server/src/modules/authorization/authorization.service.spec.ts

```diff
@@ -32,325 +24,387 @@ describe('AuthorizationService', () => {
-				const context = PermissionContextBuilder.read([]);
-				const user = userFactory.buildWithId();
-				const course = courseFactory.buildWithId();
-				const entityName = AllowedAuthorizationEntityType.Course;
-				const entityId = course.id;
-				const spyGetUserWithPermissions = jest.spyOn(service, 'getUserWithPermissions').mockResolvedValueOnce(user);
-				const spyHasPermission = jest.spyOn(service, 'hasPermission').mockReturnValue(true);
-				loader.loadEntity.mockResolvedValueOnce(course);
+				const user = userFactory.build();
+				const requiredPermissions = [Permission.ACCOUNT_CREATE];
 
-				return { context, course, user, entityName, entityId, spyGetUserWithPermissions, spyHasPermission };
-			};
+				authorizationHelper.hasAllPermissions.mockReturnValueOnce(true);
 
-			it('should call AuthorizationService.getUserWithPermissions with specific arguments', async () => {
-				const { context, user, entityName, entityId, spyGetUserWithPermissions } = setup();
+				return { user, requiredPermissions };
+			};
 
-				await service.hasPermissionByReferences(user.id, entityName, entityId, context);
+			it('should return true', () => {
+				const { user, requiredPermissions } = setup();
 
-				expect(spyGetUserWithPermissions).toBeCalledWith(user.id);
+				const result = service.hasAllPermissions(user, requiredPermissions);
 
-				spyGetUserWithPermissions.mockRestore();
+				expect(result).toBe(true);
 			});
+		});
+	});
 
-			it('should call ReferenceLoader.loadEntity with specific arguments', async () => {
-				const { context, user, entityName, entityId, spyHasPermission } = setup();
-
-				await service.hasPermissionByReferences(user.id, entityName, entityId, context);
+	describe('checkOneOfPermissions', () => {
+		describe('when hasOneOfPermissions of AuthorizationHelper returns false', () => {
+			const setup = () => {
+				const user = userFactory.build();
+				const requiredPermissions = [Permission.ACCOUNT_CREATE];
 
-				expect(loader.loadEntity).toBeCalledWith(entityName, entityId);
+				authorizationHelper.hasOneOfPermissions.mockReturnValueOnce(false);
 
-				spyHasPermission.mockRestore();
-			});
+				return { user, requiredPermissions };
+			};
 
-			it('should call AuthorizationService.hasPermission with specific arguments', async () => {
-				const { context, user, entityName, course, entityId, spyHasPermission } = setup();
+			it('should throw UnauthorizedException', () => {
+				const { user, requiredPermissions } = setup();
 
-				await service.hasPermissionByReferences(user.id, entityName, entityId, context);
+				expect(() => service.checkOneOfPermissions(user, requiredPermissions)).toThrow(UnauthorizedException);
+			});
+		});
 
-				expect(spyHasPermission).toBeCalledWith(user, course, context);
+		describe('when hasOneOfPermissions of AuthorizationHelper returns true', () => {
+			const setup = () => {
+				const user = userFactory.build();
+				const requiredPermissions = [Permission.ACCOUNT_CREATE];
```

Comment:
Eigentlich wollen wir ja in den Tests alle möglichen Parameter Kombinationen testen. So die grundsätzliche Annahme. Hier ergibt es, wenn man die Implementierung betrachtet und den authorizationHelper sowieso mockt, nicht wirklich Sinn. Aber das der User gar keine Permissions hat und das Szenario fehlschlagen würde wenn der Helper nicht gemockt wäre, finde ich auch etwas komisch. Bzw. sind auch noch weitere Parameterkombinationen denkbar, die aber mit dem gemockten Helper immer zum gleichen Ergebnis führen. Hm...
Hab nur ich das nicht ganz klar oder haben wir hier noch eine gewisse Unklarheit wie wir Tests schreiben wollen?

Reply:
Ne, guter Punkt! Das ist mir auch nicht klar.